### ISSUE TITLE
[libiconv] build failure with android ndk 26.1

### ISSUE THREAD

**Issue by LeenHawk** [microsoft/vcpkg#35315]
### Operating system

Linux Debian Sid

### Compiler

Android NDK 26.1

### Steps to reproduce the behavior

```Shell
`./vcpkg install libiconv --triplet=x64-android`
or `./vcpkg install libiconv --triplet=arm64-android`
```

### Failure logs

[config-x64-android-dbg-err.log](https://github.com/microsoft/vcpkg/files/13463595/config-x64-android-dbg-err.log)
[config-x64-android-dbg-out.log](https://github.com/microsoft/vcpkg/files/13463596/config-x64-android-dbg-out.log)
[config-x64-android-dbg-config.log](https://github.com/microsoft/vcpkg/files/13463597/config-x64-android-dbg-config.log)
[config-arm64-android-dbg-config.log](https://github.com/microsoft/vcpkg/files/13463599/config-arm64-android-dbg-config.log)
[config-arm64-android-dbg-err.log](https://github.com/microsoft/vcpkg/files/13463600/config-arm64-android-dbg-err.log)
[config-arm64-android-dbg-out.log](https://github.com/microsoft/vcpkg/files/13463601/config-arm64-android-dbg-out.log)

### Additional context

happen with android ndk 26

**Comment by dg0yt** [comment]
vcpkg doesn't support NDK r26 with autotools-based ports.
Known issue since https://github.com/microsoft/vcpkg/issues/31332.
Proposed fix in #31228 was rejected. The alternative desired by vcpkg owners hasn't arrived yet.

**Comment by jimwang118** [comment]
Thanks for posting this issue. Please reopen this issue if this is still a problem for you.
